apps/payments/next/app/[locale]/[offeringId]/[interval]/checkout/[cartId]/(otherLayout)/layout.tsx

```diff
@@ -0,0 +1,122 @@
+import { config } from 'apps/payments/next/config';
+
+export interface CheckoutSearchParams {
+  experiment?: string;
+  promotion_code?: string;
+}
+
+export default async function CheckoutOtherLayout({
+  children,
+  params,
+}: {
+  children: React.ReactNode;
+  params: CheckoutParams;
+}) {
+  const { locale } = params;
+  const acceptLanguage = headers().get('accept-language');
+  const cmsDataPromise = fetchCMSData(
+    params.offeringId,
+    acceptLanguage,
+    locale
+  );
+  const cartDataPromise = getCartAction(params.cartId);
+  const sessionPromise = auth();
+  const l10n = getApp().getL10n(acceptLanguage, locale);
+  const [cms, cart, session] = await Promise.all([
+    cmsDataPromise,
+    cartDataPromise,
+    sessionPromise,
+  ]);
+  const purchaseDetails =
+    cms.defaultPurchase.purchaseDetails.localizations.at(0) ||
+    cms.defaultPurchase.purchaseDetails;
+  console.log(cart);
+  return (
+    <MetricsWrapper cart={cart}>
+      {session?.user?.email && (
+        <div className="mb-8 tablet:hidden">
+          <SignedIn email={session.user.email} />
+        </div>
+      )}
+      <div className="mx-7 tablet:grid tablet:grid-cols-[minmax(min-content,500px)_minmax(20rem,1fr)] tablet:grid-rows-[min-content] tablet:gap-x-8 tablet:mt-4 tablet:mb-auto desktop:grid-cols-[600px_1fr]">
+        <SubscriptionTitle cartState={cart.state} l10n={l10n} />
+
+        <section
+          className="mb-6 tablet:mt-6 tablet:min-w-[18rem] tablet:max-w-xs tablet:col-start-2 tablet:col-end-auto tablet:row-start-1 tablet:row-end-3"
+          aria-label="Purchase details"
+        >
+          <PurchaseDetails
+            invoice={cart.upcomingInvoicePreview}
+            purchaseDetails={purchaseDetails}
+            priceInterval={
+              <PriceInterval
+                l10n={l10n}
+                amount={cart.upcomingInvoicePreview.listAmount}
+                currency={cart.upcomingInvoicePreview.currency}
+                interval={cart.interval}
+              />
+            }
+            totalPrice={
+              <PriceInterval
+                l10n={l10n}
+                amount={cart.upcomingInvoicePreview.totalAmount}
+                currency={cart.upcomingInvoicePreview.currency}
+                interval={cart.interval}
+              />
+            }
+          />
+          <SelectTaxLocation
+            cartId={cart.id}
+            cartVersion={cart.version}
+            cmsCountries={cms.countries}
+            locale={locale.substring(0, 2)}
+            productName={purchaseDetails.productName}
+            unsupportedLocations={config.subscriptionsUnsupportedLocations}
+            countryCode={cart.taxAddress?.countryCode}
+            postalCode={cart.taxAddress?.postalCode}
+          />
+          {cart.couponCode && (
+            <CouponForm
+              cartId={cart.id}
+              cartVersion={cart.version}
+              promoCode={cart.couponCode}
+              readOnly={true}
```

Comment:
Why opt for a new layout instead of making this conditional based on cart state?